src/index.js

```diff
@@ -0,0 +1,195 @@
+        landscape.textContent = "🌳🍂🪵🍃🦉🍁🐿️🍄🍁⛅";
+        skySelect.value = "cloudy";
+        updateSkyEmoji("cloudy");
+        sky.textContent = "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️";
+
+    } else if (tempF >= 50) {
+        tempValue.classList.add("green");
+        landscape.textContent = "🍁🌦️🌾🌤️🐛🌼🌦️🍂🪲☘️";
+        skySelect.value = "cloudy";
+        updateSkyEmoji("cloudy");
+        sky.textContent = "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️";
+
+    } else {
+        tempValue.classList.add("teal");
+        landscape.textContent = "❄️⛄🌨️🌬️⛄🌲☃️❄️🥶🌨️";
+        skySelect.value = "snowy";
+        updateSkyEmoji("snowy");
+        sky.textContent = "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨";
+    }
+}
+
+increaseButton.addEventListener("click", function () {
+    tempF++;
+    updateTempColor();
+});
+
+decreaseButton.addEventListener("click", function () {
+    tempF--;
+    updateTempColor();
+});
+updateTempColor();
+
+
+function displayCityName() {
+    cityNameInput.addEventListener("input", () => {
+        headerCityName.textContent = cityNameInput.value;
+    });
+}
+displayCityName();
+
+// retry helper function
+// wave 4: //
+const getWithRetry = (url, queryParams, attempt = 1) => {
+    return axios.get(url, {
+        params: queryParams
+    }).then((response) => {
+        const data = response.data;
+
+        if (data.error) {
+            if (attempt >= 5) {
+                console.log("Max attempts reached!");
+                return null;
+            }
+            console.log(`Empty response on attempt ${attempt}. Retrying...`);
+            return new Promise((resolve) => {
+                setTimeout(() => {
+                    resolve(getWithRetry(url, queryParams, attempt + 1));
+                }, 600 * attempt);
+            });
+        }
+        return response;
+    })
+};
+
+const findLatitudeAndLongitude = (query) => {
+    const url = "http://localhost:5000/location";
+    return getWithRetry(url, { q: query, format: 'json' }).then((response) => {
+        if (!response.data) {
+            return null;
+        }
+        const { lat: latitude, lon: longitude } = response.data[0];
+        return { latitude, longitude };
+    });
+};
+const findWeatherLatLon = (lat, lon) => {
+    const url = "http://localhost:5000/weather";
+
+    return getWithRetry(url, { lat, lon }).then((response) => {
+        if (!response.data) {
+            return null;
+        }
+        const tempK = response.data.main.temp;
+        const tempC = Math.round(tempK - 273.15);
+        const tempF = Math.round((tempC * 9) / 5 + 32);
+        return { tempF, tempC };
+    });
+};
+
+currentTempButton.addEventListener("click", () => {
+    const city = cityNameElement.textContent.trim();
+    if (!city) return;
+
+    findLatitudeAndLongitude(city)
+        .then(({ latitude, longitude }) => findWeatherLatLon(latitude, longitude))
+        .then((result) => {
+            tempF = result.tempF;
+            updateTempColor();
+        });
+});
```

Comment:
As mentioned earlier, your JS may run before the DOM is fully loaded, which could cause issues when selecting elements. To prevent this, consider wrapping your event listener registrations inside a `DOMContentLoaded` event handler. This ensures the DOM elements exist before you try to reference them.

For your event handlers, it's a good practice to define them as named functions and pass the function reference to `addEventListener`. This improves modularity by separating event registration from event logic, making your code easier to maintain and test.

**Note: This suggestion applies to all of your event registration logic.**